### ISSUE TITLE
How to decide the layer and Attention head?

### ISSUE THREAD

**Issue by Jenna-Bai** [letitiabanana/Test-Code#1]
Thanks for the sharing!

I would like to know how to decide which layer of the attention block and which attention head to prune.

In the demo, it seems to choose the 8th attention block and 9th attention head. Does this configuration fit for all models and dataset? Or it should be decided based on the performance case by case.

**Comment by letitiabanana** [comment]
Hi Jenna,

Thanks for your question!

**I would like to know how to decide which layer of the attention block and which attention head to prune.** 
-- As stated in the paper section 3.4, we designed a weakly-supervised clip based reward metric to automatically search out a well-performing layer-head combination. I would update the code for our searching asap.

**In the demo, it seems to choose the 8th attention block and 9th attention head. Does this configuration fit for all models and dataset? Or it should be decided based on the performance case by case.**
-- I performed 100 iterations of random search with our reward metric on COCO STUFF validation set and gets this layer 8, head 10 combination (it's says 9th head in the code because I count from 0) as the best performing one for COCO STUFF. **All the Blip based experimental results on all the five reported datasets utilize this layer 8 head 10 configuration.** There might be better configuration for each dataset but becauses I do not have enough resource to run our reward metric search on every dataset so I apply this layer 8 head 10 configuration to all. I think the rationale behind is that usually the layer and head VLM used to perform object localization is fixed across different images.
